modules/nf-core/pigz/compress/tests/main.nf.test

```diff
@@ -0,0 +1,33 @@
+nextflow_process {
+    name "Test Process PIGZ_COMPRESS"
+    script "../main.nf"
+    process "PIGZ_COMPRESS"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "pigz"
+    tag "pigz/compress"
+
+    test("sarscov2 - genome - fasta") {
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test'], // meta map
+                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+                    ]
+                """
+            }
+        }
+        then {
+            def lines = path(process.out.archive.get(0).get(1)).linesGzip
+            assertAll(
+                { assert process.success },
+                { assert process.out.archive != null },
+                { assert process.out.archive.get(0).get(1) ==~ ".*/genome.fasta.gz"  },
+                { assert snapshot(lines).match("uncompressed_file")},
```

Comment:
```suggestion
                { assert snapshot(path(process.out.archive[0][1])).match("uncompressed_file")},
```

can you try getting the md5 if you are not filtering....can you see the above generate a consistent md5 ?

Reply:
I tried getting the md5 of the compressed file in the [initial commit](https://github.com/nf-core/modules/pull/4560/commits/8c75bad429e250fc64055351ef072c1a872b6368#diff-a1bb306193a246481d582ca5ec5d26ac0c0bd7889efbfcda4b438766f228a893R28) and it failed. For more context, I've written the [xz/compress module](https://github.com/nf-core/modules/tree/master/modules/nf-core/xz/compress) and there simply fixing the number of `cpus` to 1 generated consistent md5s. For `pigz`, I tried fixing `cpus` to 1, and also [adding flags `--no-name --no-time`](https://github.com/nf-core/modules/pull/4560/commits/8c75bad429e250fc64055351ef072c1a872b6368#diff-07004478f8fe16325d9e66d5fd414e193483cb33834135aa8ca626447d2123afR1-R8) to try to make consistent md5. It did work on my local - through several docker, singularity and conda tests, the md5 was consistent. However, when I created this PR, [docker CI running reported inconsistent md5](https://github.com/nf-core/modules/actions/runs/7146885875/job/19465444834). As I can't reproduce on my local, I switched to snapshotting the lines themselves.

I think we can snapshot the uncompressed file in a second test by snapshotting the file after `pigz/uncompress`... will try that

Reply:
> I think we can snapshot the uncompressed file in a second test by snapshotting the file after `pigz/uncompress`... will try that

Did not find a way to do this... then looked at nf-test docs again and it seems if we simply use the `.md5` property of `path`, we will actually get the md5 of the uncompressed file contents (from https://code.askimed.com/nf-test/docs/assertions/files/):
> Note that for gzip compressed files, the md5 property is calculated after gunzipping the file contents.

However, after doing this in https://github.com/nf-core/modules/pull/4560/commits/7bc5c8e920c55fbf6a3f67656c4cd66a5b130715, it still did not work: https://github.com/nf-core/modules/actions/runs/7157383811/job/19488197673?pr=4560#step:12:41

Reply:
I switched to just checking the first and last 5 lines of the compressed file instead of the whole file in https://github.com/nf-core/modules/pull/4560/commits/8487e0ec4bbcb4f1f1abff1fac986af18d62b5cc . Could you please re-review this?

Reply:
This has been resolved, please see https://github.com/nf-core/modules/pull/4560#issuecomment-1851042044